### DISCUSSION TITLE
Smartcrop for nicer automated thumbnail generation

### DISCUSSION THREAD

**Discussion by SvenBudak** [directus/directus#5659] [Feature Requests]
I have found a really good algorithm that can estimate which detail on an image is most relevant. It would be really helpful if this feature could be activated with one click in directus.

I've been using it for a long time in another app and I'm very satisfied.

Repo: https://github.com/jwagner/smartcrop.js/

**Comment by rijkvanzanten** [comment]
That's pretty cool!

**Reply by benhaynes** [reply]
Agree. I'd be down for documenting this as an official progressive enhancement.

**Reply by SvenBudak** [reply]
I am glad that you like this featur so much that you want to implement it. Thank you! :)

**Comment by Jonesus** [comment]
Another possibility for similar functionality could be Sharp's built-in cropping strategies `entropy` or `attention` (https://sharp.pixelplumbing.com/api-resize), as Sharp is already used by Directus :eyes:

Though based on the linked docs it seems like those options are only available with `cover` cropping, not sure if that's a problem?

I think there might've been another discussion about adding Sharp's options into the Image interface, but couldn't find it for reference...

**Reply by rijkvanzanten** [reply]
> Though based on the linked docs it seems like those options are only available with cover cropping, not sure if that's a problem?

That makes perfect sense. If you'd crop with "contain", you keep the whole image within the bounding box, so you're not "cropping" within the image 🙂 

> I think there might've been another discussion about adding Sharp's options into the Image interface, but couldn't find it for reference...

The plan for now is to rework the asset presets option in project settings to accept a json object of any of the sharp options, so we can start opening up all these other options as query params on /assets ✨

**Reply by benhaynes** [reply]
It's nice to have this "POI" as an option for cropping, it is also nice to save/have it as metadata for the asset itself (for use in client-side applications). We could use this "smart" option to set a default, and allow clicking the image to set an override.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
